dev/com.ibm.ws.sipcontainer/src/com/ibm/ws/sip/stack/transport/sip/netty/SipUdpConnLink.java

```diff
@@ -419,18 +410,17 @@ public void complete() {
 	 * send thread (immediate) or from the channel thread (asynchronous)
 	 */
 	void releaseOutboundBuffer() {
-		// TODO do we need to release a buffer ?
-		// m_outboundBuffer.release();
+		// No need to release the ByteBuf as the reference count should be 0 by this point. 
+		// Reference count should have been handled by the writeandflush method
 		m_outboundBuffer = null;
 	}
 
 	public void close(Throwable e) {
-		if (!m_connected) {
-			// avoid double-close
+		// avoid double close
+		if(!closeAlreadyCalled){ 
```

Comment:
This isn't making a lot of sense to me or maybe I'm misunderstanding the variable naming. Compared to the previous close implementation, if it is not connected then it just returns otherwise if connected/open then it does the close logic of removing the listening point. Now if closeAlreadyCalled is false meaning close wasn't called then it set's it to true and returns without doing the close logic and after it is true then it can do the close logic. Am I misunderstanding?